src/later.cpp

```diff
@@ -204,8 +204,15 @@ bool execCallbacksOne(
     if (callbacks.size() == 0) {
       break;
     }
+
+#ifdef RCPP_USING_UNWIND_PROTECT // See https://github.com/r-lib/later/issues/191
+    // This line may throw errors!
+    callbacks[0]->invoke();
+#else
     // This line may throw errors!
     callbacks[0]->invoke_wrapped();
```

Comment:
Do we even need this branch at this point? Is `RCPP_USING_UNWIND_PROTECT` entirely up to us? (I think I misunderstood it as being something configured when Rcpp is installed, but my updated understanding from you both is that this is an option we decide just for ourselves in our own Makevars?)

Reply:
This is defined by the `Rcpp.h` header by default. To turn it off you'd have to define `RCPP_NO_UNWIND_PROTECT` before including the header (or as Makevars). It's true that we don't do this ourselves, but a user could still do it in their user or site Makevars. I think it'd be prudent at least for this release to keep it. That way if there are edge cases that depend on the old behaviour, they can still access this.

Reply:
In that case, my very last question (hopefully)... `invoke_wrapped()` calls `invoked()`, which now includes `unwindProtect`/`BEGIN_CPP`. Does it make more sense to have `invoked_wrapped()` continue to call a code path that leaves that out? So maybe this conditional calls either `invoke_unwind_protect()` or `invoke_wrapped()`, each of which then call `invoke()`? Or is it OK the way it is?

Reply:
It's fine the way it is as the 2 code paths are completely separate. `invoke_wrapped()` is only called when unwind protect is not defined, in which case the invoke method itself will not be wrapped in `unwindProtect`.

With unwind protect: callback->invoke | invoke including `unwindProtect/BEGIN_CPP`
No unwind protect: callback->invoke_wrapped | invoke_c | invoke (without `unwindProtect`)